client/me/form-base/with-form-base.jsx

```diff
@@ -8,96 +8,54 @@ import { connect } from 'react-redux';
 /**
  * Internal dependencies
  */
-import user from 'calypso/lib/user';
-import userSettings from 'calypso/lib/user-settings';
 import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import QueryUserSettings from 'calypso/components/data/query-user-settings';
+import getUserSettings from 'calypso/state/selectors/get-user-settings';
+import getUnsavedUserSettings from 'calypso/state/selectors/get-unsaved-user-settings';
+import hasUnsavedUserSettings from 'calypso/state/selectors/has-unsaved-user-settings';
+import {
+	setUserSetting,
+	clearUnsavedUserSettings,
+	saveUserSettings,
+} from 'calypso/state/user-settings/actions';
+import { isUpdatingUserSettings } from 'calypso/state/user-settings/selectors';
 
 const withFormBase = ( WrappedComponent ) => {
 	class EnhancedComponent extends React.Component {
 		static displayName = `withFormBase(${ WrappedComponent.displayName || WrappedComponent.name })`;
 
 		state = {
 			redirect: false,
-			submittingForm: false,
-			showNotice: false,
 		};
 
-		componentDidMount() {
-			userSettings.getSettings();
-		}
-
 		componentWillUnmount() {
 			// Silently clean up unsavedSettings before unmounting
-			userSettings.unsavedSettings = {};
-		}
-
-		componentDidUpdate() {
-			this.showNotice();
+			this.props.clearUnsavedUserSettings();
 		}
 
 		getDisabledState = () => {
-			return this.state.submittingForm;
+			return this.props.isUpdatingUserSettings;
 		};
 
-		showNotice() {
-			if ( userSettings.initialized && this.state.showNotice ) {
-				this.props.successNotice( this.props.translate( 'Settings saved successfully!' ), {
-					id: 'form-base',
-				} );
-				this.state.showNotice = false;
-			}
-		}
-
 		getSetting = ( settingName ) => {
-			return userSettings.getSetting( settingName ) || '';
+			const { _unsavedUserSettings, _userSettings } = this.props;
+			return _unsavedUserSettings[ settingName ] ?? _userSettings[ settingName ] ?? '';
 		};
 
 		toggleSetting = ( event ) => {
```

Comment:
Is this one used or planned to be used anywhere? Can we ditch everything we're not using?

Reply:
It'll be used in another component which at the moment still uses the form base mixin.